libsession/src/main/java/org/session/libsession/messaging/messages/control/DataExtractionNotification.kt

```diff
@@ -0,0 +1,81 @@
+package org.session.libsession.messaging.messages.control
+
+import com.google.protobuf.ByteString
+import org.session.libsignal.libsignal.ecc.ECKeyPair
+import org.session.libsignal.service.internal.push.SignalServiceProtos
+import org.session.libsignal.utilities.logging.Log
+import java.lang.Exception
+
+class DataExtractionNotification(): ControlMessage() {
+    var kind: Kind? = null
+
+    // Kind enum
+    sealed class Kind {
+        class Screenshot() : Kind()
+        class MediaSaved(val timestanp: Long) : Kind()
+
+        val description: String = run {
+            when(this) {
+                is Screenshot -> "screenshot"
+                is MediaSaved -> "mediaSaved"
+            }
+        }
+    }
+
+    companion object {
+        const val TAG = "DataExtractionNotification"
+
+        fun fromProto(proto: SignalServiceProtos.Content): DataExtractionNotification? {
+            val dataExtractionNotification = proto.dataExtractionNotification ?: return null
+            val kind: Kind
+            when(dataExtractionNotification.type) {
+                SignalServiceProtos.DataExtractionNotification.Type.SCREENSHOT -> kind = Kind.Screenshot()
+                SignalServiceProtos.DataExtractionNotification.Type.MEDIA_SAVED -> {
+                    val timestamp = if (dataExtractionNotification.hasTimestamp()) dataExtractionNotification.timestamp else 0
```

Comment:
Let's return `null` here if the timestamp is missing, because a `DataExtractionNotification` of type `MediaSaved` without a valid timestamp is invalid :)